src/grid/styled-components.js

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright (c) 2018-2019 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import React from 'react';
+import {styled} from '../styles/index.js';
+import {getMediaQueries} from '../helpers/responsive-helpers.js';
+import {BEHAVIOR} from './constants.js';
+
+import type {
+  SoftResponsiveT,
+  GridPropsT,
+  CellPropsT,
+  StyledGridPropsT,
+  StyledCellPropsT,
+} from './types.js';
+
+// defaults
+const GRID_COLUMNS = [4, 8, 12];
+const GRID_GUTTERS = [16, 36, 36];
+const GRID_MARGINS = [16, 36, 64];
+const GRID_MAX_WIDTH = 1280;
+
+export function Grid({
+  align,
+  behavior,
+  children,
+  gridColumns,
+  gridGaps,
+  gridGutters,
+  gridMargins,
+  gridMaxWidth,
+}: GridPropsT) {
+  return (
+    <StyledGrid
+      $behavior={behavior}
+      $gridMargins={gridMargins}
+      $gridGutters={gridGutters}
+      $gridMaxWidth={gridMaxWidth}
+      $align={align}
+    >
+      {React.Children.map(children, child =>
+        React.cloneElement(child, {
+          $gridColumns: gridColumns,
+          $gridGutters: gridGutters,
+          $gridGaps: gridGaps,
+        }),
+      )}
+    </StyledGrid>
+  );
+}
+
+export const StyledGrid = styled<StyledGridPropsT>(
+  'div',
+  ({
+    $align,
+    $behavior = BEHAVIOR.fixed,
+    $gridGutters = GRID_GUTTERS,
+    $gridMargins = GRID_MARGINS,
+    $gridMaxWidth = GRID_MAX_WIDTH,
+    $theme,
+  }) => {
+    const mediaQueries = getMediaQueries($theme.breakpoints);
+    const gridStyles = mediaQueries.reduce(
+      (acc, cur, idx) => {
+        return {
+          ...acc,
+          [cur]: {
+            paddingLeft: `${getResponsiveNumber($gridMargins, idx) -
+              getResponsiveNumber($gridGutters, idx) / 2 -
+              0.5}px`,
+            paddingRight: `${getResponsiveNumber($gridMargins, idx) -
+              getResponsiveNumber($gridGutters, idx) / 2 -
+              0.5}px`,
+            alignItems: getResponsiveValue($align, idx),
+          },
+        };
+      },
+      {
+        paddingLeft: `${getResponsiveNumber($gridMargins, 0) -
+          getResponsiveNumber($gridGutters, 0) / 2 -
+          0.5}px`,
+        paddingRight: `${getResponsiveNumber($gridMargins, 0) -
+          getResponsiveNumber($gridGutters, 0) / 2 -
+          0.5}px`,
+        alignItems: getResponsiveValue($align, 0),
+      },
+    );
+    return {
+      boxSizing: 'border-box',
+      display: 'flex',
+      flexWrap: 'wrap',
+      marginLeft: 'auto',
+      marginRight: 'auto',
+      maxWidth:
+        $behavior === BEHAVIOR.fixed
+          ? `${$gridMaxWidth +
+              2 * getResponsiveNumber($gridMargins, Infinity) -
+              1}px`
+          : null,
+      ...gridStyles,
+    };
+  },
+);
+
+export function Cell({
```

Comment:
Same for the `Cell` component, please move it out of the `styled-components.js`.